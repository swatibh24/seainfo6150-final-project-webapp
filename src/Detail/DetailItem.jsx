import React from "react";
import { Link } from "react-router-dom";
import styles from "./DetailItem.module.css";
import PropTypes from "prop-types";

const DetailItem = (props) => {
    //const recipe=props.recipe;
    return(
      <div className={styles.recipeContainer}>
        <div className={styles.Recipe_part}>
        <img
          className={styles.recipeImg}
          src={props.recipe.imageURL}
          alt={props.recipe.name}
        />
        <h1 className={styles.recipeName}>{props.recipe.name}</h1>
        <p className={styles.recipeDescription}>{props.recipe.description}</p>
        <h2 className={styles.recipeTitle}>Ingredients</h2>
        <p className={styles.recipeIngredients}>{props.recipe.ingredients}</p>
        <h2 className={styles.recipeTitle}>Cooking Directions</h2>
        <p className={styles.recipeDetail}>{props.recipe.one}</p>
        <p className={styles.recipeDetail}>{props.recipe.two}</p>
        <p className={styles.box}>
          <Link className={styles.catBackLink} to={props.recipe.categoryURL}>
            &lt; Back to {props.recipe.categoryName} Page
          </Link>
          <Link className={styles.recipeBackLink} to="/recipes">
        Back to All Recipes Page &gt;
      </Link>
        </p>
        </div>
      </div>
    );
    
};

DetailItem.propTypes = {
  recipe: PropTypes.object.isRequired,
};

export default DetailItem;