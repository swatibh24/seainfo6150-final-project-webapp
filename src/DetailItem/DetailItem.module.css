.recipeContainer {
    max-width: 60rem;
    min-width: 20em;
    margin: auto;
    padding: 1px;
    background-color:#F5F5F5;
    font-size:20px;
    width:80%;
    height:auto;
    }
  
    .Recipe_part {
      margin-top:50px;
      margin-bottom:50px;
    }

     .linkBox{
          margin-bottom:20px;
        }
  
    .Recipe_part img {
      width: 60%;
      float:inherit;
      margin-left: 20%;
      height:auto;
    
    }
 .recipesBackLink {
    text-decoration: none;
    color: black;
    font-family: Futura, Arial, sans-serif;
  }
  
    .Recipe_part p {
      padding: 1px;
      margin-left: 20%;
      background-color: #F5F5F5;
      font-size:15px;
      border:1px solid #F5F5F5;
      width:60%;
      height:auto;
      text-align: left;
    
    }
    
    .recipeImg {
      margin-bottom: 30px;
      border-radius: 10px;
    }
    
    .recipeName {
      margin-bottom: 30px;
      font-weight: bold;
      font-family: Tahoma;
      text-align:center;
    }
    
    .recipeTitle {
      font-weight: bold;
      font-family: Verdana;
      text-align:center;
    }
    
    .recipeIngredients {
      margin-bottom: 30px;
      font-family: Verdana, Helvetica, sans-serif;
      text-align:left;
    }
  
    .recipeDescription{
      font-family:sans-serif;
      font-style: italic;
      font-size:15px;
      color:gray;
    }
    
    .recipeDetail {
      font-family: Verdana, Helvetica, sans-serif;
      text-align:left;
    }
    
    .catBackLink {
      text-decoration: none;
      color: rgb(49, 105, 228);
      text-align: left;
      font-family: Futura, Arial, sans-serif;
    }
  .box{
      width: 100%;
      height: 30px;
      display: grid;
      flex-flow: column-reverse;
      justify-content: space-around;
      align-items: stretch;
      grid-template-columns: repeat(2, 30fr);
  }
    .recipeBackLink {
          text-decoration: none;
          color: rgb(49, 105, 228);
          text-align: right;
          font-family: Futura, Arial, sans-serif;
        }
    @media screen and (min-width: 400px) {
      .recipeImg {
        width: 400px;
      }
      .recipeName {
        font-size: 20px;
      }
      
      .recipeBackLink,
      .catBackLink{
        font-size: 9px;
      }

      .recipeTitle{
        font-size:15px;
      }
      .recipeDetail,
      .recipeIngredients {
        font-size: 12px;
      }
    }
  
    @media screen and (min-width: 600px) {
      .recipeImg {
        width: 400px;
      }
      .recipeName {
        font-size: 20px;
      }
      
      .recipeBackLink,
      .catBackLink {
        font-size: 13px;
      }

      .recipeTitle{
        font-size:15px;
      }
      .recipeDetail,
      .recipeIngredients {
        font-size: 12px;
      }
    }
    
    @media screen and (min-width: 900px) {
      .recipeImg {
        width: 500px;
      }
      .recipeName {
        font-size: 22px;
      }
      .recipeTitle,
      .recipeBackLink,
      .catBackLink {
        font-size: 18px;
      }
      .recipeDetail,
      .recipeIngredients {
        font-size: 14px;
      }
    }
    
    @media screen and (min-width: 1200px) {
      .recipeImg {
        width: 600px;
      }
      .recipeName {
        font-size: 25px;
      }
      .recipeTitle,
      .recipeBackLink,
      .catBackLink {
        font-size: 20px;
      }
      .recipeDetail,
      .recipeIngredients {
        font-size: 15px;
      }
    }