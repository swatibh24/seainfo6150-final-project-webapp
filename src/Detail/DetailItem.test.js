import React from "react";
import { render, fireEvent } from "@testing-library/react";
import DetailItem from "./DetailItem.jsx";
import { Link, BrowserRouter } from "react-router-dom";

it("renders the DetailItem component correctly", () => {
  const recipe = {
    "Cream of Mushroom Soup": {
      "categoryID": "1",
      "recipeID": "11",
      "ID":"11-Cream-of-Mushroom-soup",
      "categoryName": "Soup",
      "name": "Cream of Mushroom Soup",
      "description": "A simple type of soup where a basic roux is thinned with cream or milk and then mushrooms and/or mushroom broth are added.",
      "ingredients": "2 tablespoons butter, 1/2 pound sliced fresh mushrooms, 1/4 cup chopped onion, 6 tablespoons all-purpose flour, 1/2 teaspoon salt, 1/8 teaspoon pepper, 2 cans (14-1/2 ounces each) chicken broth, 1 cup half-and-half cream",
        "one": "Step 1: In a large saucepan, heat butter over medium-high heat; saute mushrooms and onion until tender.",
        "two": "Step 2: Mix flour, salt, pepper and 1 can broth until smooth; stir into mushroom mixture. Stir in remaining broth. Bring to a boil; cook and stir until thickened, about 2 minutes. Reduce heat; stir in cream. Simmer, uncovered, until flavors are blended, about 15 minutes, stirring occasionally.",
      "imageURL": "https://www.budgetbytes.com/wp-content/uploads/2017/09/Creamy-Garlic-Mushroom-Soup-spoon.jpg",
      "detailURL": "/recipe/11",
      "categoryURL": "/category/1"
    },
  }  
  const { container } = render(<BrowserRouter><DetailItem recipe={recipe}/></BrowserRouter>);
  expect(container).toMatchSnapshot();
});