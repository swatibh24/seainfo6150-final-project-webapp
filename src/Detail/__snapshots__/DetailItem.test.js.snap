// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DetailItem component correctly 1`] = `
<div>
  <div
    class="recipeContainer"
  >
    <div
      class="Recipe_part"
    >
      <img
        class="recipeImg"
      />
      <h1
        class="recipeName"
      />
      <p
        class="recipeDescription"
      />
      <h2
        class="recipeTitle"
      >
        Ingredients
      </h2>
      <p
        class="recipeIngredients"
      />
      <h2
        class="recipeTitle"
      >
        Cooking Directions
      </h2>
      <p
        class="recipeDetail"
      />
      <p
        class="recipeDetail"
      />
      <p
        class="box"
      >
        <a
          class="catBackLink"
          href=""
        >
          &lt; Back to 
           Page
        </a>
        <a
          class="recipeBackLink"
          href="/recipes"
        >
          Back to All Recipes Page &gt;
        </a>
      </p>
    </div>
  </div>
</div>
`;
